packages/easysearch:core/lib/engines/external-index.js

```diff
@@ -95,20 +109,26 @@ class ExternalEngine extends ReactiveEngine {
     const selObj = this.callConfigMethod(
         'selector',
         searchDefinition,
-        options,
-      this.config.aggregation),
+        options),
       findOptions = this.getFindOptions(searchDefinition, options);
     const selector = selObj.selector;
     const searchString = selObj.searchString;
     check(searchString, String);
     check(options, Object);
-    check(selector, Object);
+    //check(selector, Object);
     check(findOptions, Object);
-    this.callConfigMethod('externalFetch', searchString, options, this.col);
+    const time1 = new Date().getTime();
+    const fetchedData = this.externalFetch(selector, searchString, findOptions);
+    const time2 = new Date().getTime();
+    const preparedSelector = this.prepareData(fetchedData);
+    const time3 = new Date().getTime();
+    console.log('Search Took:', time3-time1, time3 - time2);
+    const collection = options.index.collection;
+    console.log('preparedSelector', preparedSelector);
     return new Cursor(
-      this.col.find(selector, findOptions),
-      this.col.find(selector).count()
-    );
+      collection.find(preparedSelector),
```

Comment:
Another issue I noticed is that using $in operator does not preserve order, therefore we will need to fetch results manually from DB, insert them to this.col (memory collection) with field _searchSort setup based on the order returned from backend and then return cursor to this collection (sorted by _searchSort).

something like this: this.col.insert({...doc, _searchSort: index});
and then this: this.col.find(preparedSelector, {sort: {_searchSort: 1}})